ls1intum/Artemis
# Consistent use of DTOs in Jira/Bitbucket/BambooService and improved test coverage in BitbucketService

### Description

This add test cases for the BitbucketService and handles edge cases by catching Exceptions.
The methods deleteWebHook() and deleteExistingWebHooks() were unused and therefore deleted.

Also this PR gets rid of the Map Casting in the restTemplate and its mock functions by replacing them with DTOs.

### Steps for Testing

As now all rest calls for Bamboo/Bitbucket/Jira are called with DTOs, all relevant rest calls should be tested. 

1. Login using a TUM account over JIRA
2. Create a Programming Exercise and verify that all necessary steps are working as expected.
3. Participate in it and also verify that all works as expected.
4. Delete the exercise and make sure everything works
5. Use an admin account to change the groups of an existing user in the user management view


### Test Coverage

* BitbucketService.java: 51% -> 76%
